tests/Kconfig

```diff
@@ -16,20 +16,48 @@ config TEST
 
 config TEST_EXTRA_STACKSIZE
 	int "Test function extra thread stack size"
+	default 1024 if COVERAGE_GCOV
 	default 768 if XTENSA
 	default 0
 	help
 	  Additional stack for tests on some platform where default is not
 	  enough.
 
+config HAS_COVERAGE_SUPPORT
+	bool
+	default n
+	help
+	  The code coverage report generation is a function of available RAM.
+	  Hence only available for certain boards.
+
+config PREVENT_COVERAGE
+	bool "Prevent GCOV code from getting built"
+	default n
+	help
+	  The code coverage report generation is prevented even if COVERAGE
+	  is set to true. Needed for tests which shoudn't be compiled with
+	  coverage option.
+
 config COVERAGE
 	bool "Create coverage data"
-	depends on NATIVE_APPLICATION
+	depends on HAS_COVERAGE_SUPPORT
+	depends on !PREVENT_COVERAGE
+	default n
 	help
 	  This option will build your application with the -coverage option
 	  which will generate data that can be used to create coverage reports.
 	  Currently this is fully supported only on the native POSIX port.
 
+if COVERAGE
+config COVERAGE_GCOV
+	bool "Create Coverage data from hardware platform"
+	depends on !NATIVE_APPLICATION
+	default y
+	help
+	  This option will select the custom gcov library. The reports will
+	  be available over serial .
```

Comment:
It needs to be discoverable that the user can run gen_gcov_files.py on the log output.

This could be discoverable in several ways, it could be output in the log output, it could be in some coverage .rst documentation, or it could be mentioned in the Kconfig help text.

Also, users should be presented with a possible way to read the .gcna files. E.g. somewhere it should be mentioned something like;

.gcna files can be read by the lcov utility, see "lcov homepage" for details.